src/khoj/routers/auth.py

```diff
@@ -108,10 +111,12 @@ async def auth(request: Request):
     except OAuthError as error:
         return HTMLResponse(f"<h1>{error.error}</h1>")
     khoj_user = await get_or_create_user(idinfo)
+
     if khoj_user:
         request.session["user"] = dict(idinfo)
 
-        if not khoj_user.last_login:
+        if datetime.timedelta(minutes=10) > (datetime.datetime.now(datetime.UTC) - khoj_user.date_joined):
+            asyncio.create_task(send_welcome_email(idinfo["name"], idinfo["email"]))
```

Comment:
`last_login` wasn't actually populating correctly, so `date_joined` is more accurate. 10 minutes' delta is just to be safe in case there's a lag in the code for some reason. I can reduce it to like 3 to the same effect.